rust/xaynet-analytics/src/data_combiner/data_combiner.rs

```diff
@@ -0,0 +1,235 @@
+use chrono::{DateTime, Datelike, Duration, NaiveDate, Utc};
+use std::time::UNIX_EPOCH;
+
+use crate::data_combiner::data_points::data_point::{
+    Calculate, DataPoint, DataPointMetadata, Period, PeriodUnit,
+};
+use crate::data_combiner::data_points::screen_active_time::ScreenActiveTime;
+use crate::data_combiner::data_points::screen_enter_count::ScreenEnterCount;
+use crate::data_combiner::data_points::was_active_each_past_period::WasActiveEachPastPeriod;
+use crate::data_combiner::data_points::was_active_past_n_days::WasActivePastNDays;
+use crate::repo::analytics_event::AnalyticsEvent;
+use crate::repo::repo::Repository;
+
+pub struct DataCombiner<R> {
+    repo: Box<R>,
+}
+
+impl<R> DataCombiner<R>
+where
+    R: Repository,
+{
+    pub fn new(repo: Box<impl Repository>) -> DataCombiner<impl Repository> {
+        DataCombiner { repo }
+    }
+
+    pub fn init_data_points(&self) -> Vec<DataPoint> {
+        [
+            self.to_data_points(self.init_was_active_past_n_days_vars()),
+            self.to_data_points(self.init_screen_active_time_vars()),
+            self.to_data_points(self.init_screen_enter_count_vars()),
+            self.to_data_points(self.init_was_active_each_past_period_vars()),
+        ]
+        .concat()
+    }
+
+    fn to_data_points(&self, calculables: Vec<impl Calculate>) -> Vec<DataPoint> {
+        calculables
+            .into_iter()
+            .map(|calculable| DataPoint::new(calculable.metadata(), calculable.calculate()))
+            .collect()
+    }
+
+    fn init_was_active_past_n_days_vars(&self) -> Vec<impl Calculate> {
+        let metadatas = vec![
+            DataPointMetadata {
+                period: Period {
+                    unit: PeriodUnit::Days,
+                    n: 1,
+                },
+                end: Utc::now(),
```

Comment:
will do about `Utc::now()`.

regarding dates, the timestamps from `AnalyticsEvent` are going to be DateTime for sure, i would say, so we need to either convert them and compare to a Date, or keep them and use a DateTime here. what makes more sense to you? i thought a DateTime here is more flex

Reply:
Then what we can is to have a method `new(Date, times: u32)` on `Period` and then inside that method with set it to the midnight of that day converting it to a `DateTime` in this way we are sure by construction that each `end` in a `Period` is in form we want. does it make sense?

Reply:
yeh i see what you mean.

the thing is that maybe in the future we may want to support sending data more frequently than daily (maybe it's not the best for the PET etc, but it should be supported, i guess), which should happen via some config or so. then maybe neither solution is ok, since the current implementation hardcodes getting the midnight, but also converting it to `Date` won't work.

but i totally see your point about having the data in the format we need it and simplify stuff. 

is it ok if we postpone this for a later point when we have a better idea of what needs to be supported? requirements are still a bit fuzzy right now